src/frame_compression.jl

```diff
@@ -215,3 +215,7 @@ function TranscodingStreams.process(codec::LZ4Decompressor, input::Memory, outpu
 
 end
 
+Base.@deprecate_binding LZ4Compressor LZ4FrameCompressor
+Base.@deprecate_binding LZ4Decompressor LZ4FrameDecompressor
+Base.@deprecate_binding LZ4CompressorStream LZ4FrameCompressorStream
+Base.@deprecate_binding LZ4DecompressorStream LZ4FrameDecompressorStream
```

Comment:
It's usually helpful to put deprecations into a separate file so they aren't forgotten about. Additionally adding a version comment stating when these were introduced is useful for determining when they can be removed.